### ISSUE TITLE
Redis cache grows unilimmited

### ISSUE THREAD

**Issue by toxix** [magento/magento2#25487]
# Preconditions (*)
1. Magento 2.3.3
2. Redis Cache enabled
3. Production mode

### Steps to reproduce (*)

1. Run Magento in Production with 15 Stores and organic trafic
2. Redis caching enabled

### Expected result (*)

1. Redis cache will stabilize in size.
2. Cache key expiery is set for all things that do have an expiery

### Actual result (*)

1. Redis is growing in size to unlimited (>30GB)
2. Cache key expirey is not set
```
> info keyspace
# Keyspace
db0:keys=45775,expires=41711,avg_ttl=79567831
db1:keys=49089,expires=27837,avg_ttl=16355872
db2:keys=706,expires=706,avg_ttl=696476
```

## Further information
The only way to prevent our production system server from dying was to set a hard limit for Redis memory consumption. The relying on the memory limit of redis has some performance impact and there are always things cleared that don't need to and vice versa. This should be handled by the application with setting an resonable ttl.

# fix
This is actualy an known bug and already long time fixed in an dependency of magento core composer file. See also https://github.com/colinmollenhour/Cm_Cache_Backend_Redis/commit/bc63e7226a525fe95865c756aad0037f0d57d03c
But this is not included in current magento version Last update of this core depedency was in September 2018: https://github.com/magento/magento2/commit/32058c748731348e7d38cffa7a6c56113e17b3d8

Applying this change as a hotfix, actualy stabalize our production redis at 3,5GB ram usage, instead of growing over 30GB. And also boost the performance of the system quiete a bit.

## Hot Fix
Add the following requirement to your project composer.json file:
```
    "colinmollenhour/cache-backend-redis": "1.11.0 as 1.10.6"
```

## Fix
I will File a PR with an updated composer dependency to the magento core and reference this issue.

**Comment by PascalBrouwers** [comment]
@toxix Having the same problem but only at certain periods it will grow unlimited. I will see if your fix works too.

**Comment by kilis** [comment]
Same is also on Magento version 2.1.7. Did install only that fix, but nothing has really changed.

**Comment by magento-engcom-team** [comment]
Hi @toxix. Thank you for your report. 
The issue has been fixed in magento/magento2#25488 by @toxix in 2.4-develop branch
Related commit(s):
 - [49809a7af6493851d88e858cf5b7dcead56ec38e](https://github.com/magento/magento2/commit/49809a7af6493851d88e858cf5b7dcead56ec38e)
 - [d4931eca33116a8dddba85ef6b139efa75b8bb5b](https://github.com/magento/magento2/commit/d4931eca33116a8dddba85ef6b139efa75b8bb5b)
 - [02edd9d1196659e6204c7839d400fbb4542b48c2](https://github.com/magento/magento2/commit/02edd9d1196659e6204c7839d400fbb4542b48c2)
 - [3b89fc4e0d5e6383ad05b1904ba1e3ccafd3950c](https://github.com/magento/magento2/commit/3b89fc4e0d5e6383ad05b1904ba1e3ccafd3950c)
 - [85e71f5745447fa083dd00b9f197fb6f49dd2327](https://github.com/magento/magento2/commit/85e71f5745447fa083dd00b9f197fb6f49dd2327)
 - [a8f1ee7a56fcb13496b4ea70dd906c61d66685e9](https://github.com/magento/magento2/commit/a8f1ee7a56fcb13496b4ea70dd906c61d66685e9)
 - [566c625ff5fbef7cabd64a6b9f9a78a15e857c4d](https://github.com/magento/magento2/commit/566c625ff5fbef7cabd64a6b9f9a78a15e857c4d)
 - [6d0fa267f2950dc6243a312d7a48d4b967be770e](https://github.com/magento/magento2/commit/6d0fa267f2950dc6243a312d7a48d4b967be770e)

The fix will be available with the upcoming 2.4.0 release.
